tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/type-test-expr.bal

```diff
@@ -536,6 +536,54 @@ function testObjectIsCheckWithCycles() {
     assertFalse(a3 is object { Quuz f; });
 }
 
+service class ServiceClassA {
+    remote function x() {
+    }
+}
+
+service class ServiceClassB {
+}
+
+service class ServiceClassC {
+    resource function get hello(string name) returns string {
+        return "Hello, " + name;
+    }
+}
+
+service class ServiceClassD {
+    remote function x() {
+        int a = 0;
+        float b = 0.0;
+    }
+
+    resource function get weight() returns float {
+        return 123.4;
+    }
+}
+
+function testServiceObjects() {
+    any a = new ServiceClassA();
+    any b = new ServiceClassB();
+    any c = new ServiceClassC();
+    any d = new ServiceClassD();
+
+    assertTrue(a is ServiceClassA);
```

Comment:
Shall we check the `ServiceClassB` & `ServiceClassC` as well for variables `a` & `d`?

Reply:
I have not added them yet because I have this question, https://github.com/ballerina-platform/ballerina-lang/issues/31942#issuecomment-890961115. I will add them after I get the answer for this. Currently the runtime behaves as same as the compile time. When there is no compile time error the `is` check will give `true` otherwise `false`.